WordPress/Sniffs/PHP/IniSetSniff.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\PHP;
+
+use WordPressCS\WordPress\AbstractFunctionParameterSniff;
+use \PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Checks use of ini_set function with a blacklist for errors.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   x.x.x
+ */
+class IniSetSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * Array of functions that must be checked
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $target_functions = array(
+	 *         (string) Function name.
+	 *     );
+	 */
+	protected $target_functions = array(
+		'ini_set' => array(),
+	);
+
+	/**
+	 * Array of options that are allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $whitelisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'valid_values' = array()
+	 *         )
+	 *     );
+	 */
+	protected $whitelisted_options = array(
+		'auto_detect_line_endings' => array(),
+		'highlight.bg'             => array(),
+		'highlight.comment'        => array(),
+		'highlight.default'        => array(),
+		'highlight.html'           => array(),
+		'highlight.keyword'        => array(),
+		'highlight.string'         => array(),
+		'short_open_tag'           => array(
+			'valid_values' => array( 'true', '1', 'On' ),
+		),
+	);
+
+	/**
+	 * Array of options that are not allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $blacklisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'invalid_values' = array()
+	 *             (string) 'message'
+	 *         )
+	 *     );
+	 */
+	protected $blacklisted_options = array(
+		'max_execution_time' => array(
+			'message' => 'Use `set_time_limit()` instead.',
+		),
+		'short_open_tag' => array(
+			'invalid_values' => array( 'false', '0', 'Off' ),
+			'message'        => 'Turning off short_open_tag is prohibited as it might possibily break other plugins.',
+		),
+		'bcmath.scale' => array(
+			'message' => 'Use `bcscale()` instead.',
+		),
+		'display_errors' => array(
+			'message' => 'Use `WP_DEBUG_DISPLAY` instead.',
+		),
+		'error_reporting' => array(
+			'message' => 'Use `WP_DEBUG` instead.',
+		),
+		'filter.default' => array(
+			'message' => 'Use the filter flag constants when calling the functions instead (as you will possibly break other plugins if you change this).',
```

Comment:
For this and the one below it:

```suggestion
			'message' => 'Changing the option value can break other plugins. Use the filter flag constants when calling the Filter functions instead.',
```

Reply:
This same change still needs to be made for the `filter.default_flags` message.